aten/src/ATen/native/cuda/ForeachPointwiseOp.cu

```diff
@@ -74,7 +73,37 @@ void foreach_tensor_##NAME##_cuda_(TensorList input, TensorList tensors1, Tensor
     foreach_pointwise_op_<OP>(input, tensors1, tensors2, scalar);                                                              \
 }
 
-FOREACH_UNARY_OP(addcmul, std::multiplies);
-FOREACH_UNARY_OP(addcdiv, std::divides);
+FOREACH_POINTWISE_OP(addcmul, std::multiplies);
+FOREACH_POINTWISE_OP(addcdiv, std::divides);
+
+std::vector<Tensor> foreach_tensor_max_cuda(TensorList tensors1, TensorList tensors2) {
+    TORCH_CHECK(tensors1.size() > 0, "Tensor list must have at least one tensor.");
+    TORCH_CHECK(tensors1.size() ==  tensors2.size(), "Tensor lists must be of the same length.");
+    
+    if (!can_use_fast_route(tensors1, tensors2)) {
+        return at::native::foreach_tensor_max_slow(tensors1, tensors2);
+    }
+
+    std::vector<std::vector<at::Tensor>> tensor_lists;
+    std::vector<at::Tensor> vec_res;
+    vec_res.reserve(tensors1.size());
+    for (const auto& t: tensors1) {
+        vec_res.emplace_back(at::native::empty_like(t));
+    }
+
+    tensor_lists.emplace_back(tensors1.vec());
+    tensor_lists.emplace_back(tensors2.vec());
+    tensor_lists.emplace_back(std::move(vec_res));
+    
+    AT_DISPATCH_ALL_TYPES_AND(kHalf, tensors1[0].scalar_type(), "foreach_pointwise_op_cuda", [&]() {
+        using opmath_t = get_opmath_t<scalar_t>::opmath_t;
+        auto op = []  GPU_LAMBDA (opmath_t a, opmath_t b) -> opmath_t { return std::max(a, b); };
```

Comment:
I'm afraid std::max won't properly propagate nans, and you want behavior to agree with eager mode. Please see a functor that eager max uses for it. Would also be good to add tests with nans.

Reply:
Added two [tests](https://github.com/pytorch/pytorch/pull/45692/files#diff-b300fed36e386acda9146c82f11d6090R199) and they are passing. am i missing something?

Reply:
Yeah, using std::min/std::max is unsafe, see e.g. here https://stackoverflow.com/questions/39919069/why-does-release-debug-have-a-different-result-for-stdmin. Tests might be passing in some circumstances and failing in others.